spatialscaper/core.py

```diff
@@ -661,6 +709,28 @@ def synthesize_events_and_labels(self, all_irs, all_ir_xyzs, out_audio):
 
         return out_audio, labels
 
+    def get_background_noise(self, out_audio):
+        for ievent, event in enumerate(self.bg_events):
+            if not event.source_file:
+                ambient = self.generate_noise(event)
+            else:
+                if event.source_time is not None:
+                    ambient, _ = librosa.load(
+                        event.source_file,
+                        sr=self.sr,
+                        offset=event.source_time,
+                        duration=event.event_duration,
+                    )
+                else:  # repeat ambient file until scape duration
```

Comment:
As far as I can tell this considers background noise duration < scape duration. Maybe we should handle the case where the noise is longer, for which we should pick a random portion of the noise of duration len(scape)

Reply:
the lines right above this `else` statement do exactly that. Using `librosa.load` with `offset` and `duration`, where offset is randomly and uniformly sampled from the entire duration of the ambient noise file, and duration is the scape duration.